flowauth/frontend/cypress/integration/login_spec.js

```diff
@@ -9,3 +13,59 @@ describe("Login", function() {
 		cy.getCookie("X-CSRF").should("exist");
 	});
 });
+
+describe("Login_error_username", function () {
+	it("Fail to log in with incorrect username", function () {
+		cy.visit("/");
+		// Attempt to log in with incorrect username
+		cy.get("#username").type("WRONG_USER");
+		cy.get("#password").type("DUMMY_PASSWORD");
+		cy.get("button").click();
+		// Check that dialog appears with correct title and description
+		cy.get("#error-dialog-title").should("contain", "Error");
+		cy.get("#error-dialog-description").should("contain", "Incorrect username or password.");
+	});
+});
+
+describe("Login_error_password", function () {
+	it("Fail to log in with incorrect password", function () {
+		cy.visit("/");
+		// Attempt to log in with incorrect password
+		cy.get("#username").type("TEST_USER");
+		cy.get("#password").type("WRONG_PASSWORD");
+		cy.get("button").click();
+		// Check that dialog appears with correct title and description
+		cy.get("#error-dialog-title").should("contain", "Error");
+		cy.get("#error-dialog-description").should("contain", "Incorrect username or password.");
+	});
+});
+
+describe("Login_error_twice", function () {
+	it("Error dialog appears more than once", function () {
```

Comment:
Very minor note, but I had to read comments inside the tests to understand what the description meant. How about rephrasing it along the following lines?
```suggestion
	it("Error dialog re-appears after closing if password wasn't changed", function () {
```